crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -34,13 +37,49 @@ impl InternalVar {
             || self.expression == b.expression
     }
 
+    pub fn default() -> InternalVar {
+        InternalVar {
+            expression: Arithmetic::default(),
+            witness: None,
+            idx: None,
+        }
+    }
+
     fn new(expression: Arithmetic, witness: Option<Witness>, id: Index) -> InternalVar {
         InternalVar {
             expression,
             witness,
             idx: Some(id),
         }
     }
+
+    pub fn to_const(&self) -> Option<FieldElement> {
+        if self.expression.mul_terms.is_empty() && self.expression.linear_combinations.is_empty() {
+            return Some(self.expression.q_c);
+        }
+        None
+    }
+}
+
+impl From<&Arithmetic> for InternalVar {
+    fn from(arith: &Arithmetic) -> InternalVar {
+        //TODO shouldn't we rather move arith?
```

Comment:
Done.
Sorry for the newbie question, but using `fn from(arith: Arithmetic)` will move the `arith` argument to the `IntervalVar` object, so that no copy is done, is my understanding correct?

Reply:
That is correct